packages/starknet-snap/src/state/transaction-state-manager.ts

```diff
@@ -0,0 +1,305 @@
+   *
+   * @param param - The param object.
+   * @param param.txnHash - The txn hash of the transaction object to search for.
+   * @param [param.chainId] - The optional chain id of the transaction object to search for.
+   * @param [state] - The optional SnapState object.
+   * @returns A Promise that resolves with the transaction object if found, or null if not found.
+   */
+  async findTransaction(
+    {
+      txnHash,
+      chainId,
+    }: {
+      txnHash: string;
+      chainId?: string;
+    },
+    state?: SnapState,
+  ): Promise<Transaction | null> {
+    const filters: ITxFilter[] = [new TxHashFilter([txnHash])];
+    if (chainId !== undefined) {
+      filters.push(new ChainIdFilter([chainId]));
+    }
+    return this.find(filters, state);
+  }
+
+  /**
+   * Updates a transaction object in the state with the given data.
+   *
+   * @param data - The updated transaction object.
+   * @returns A Promise that resolves when the update is complete.
+   * @throws {StateManagerError} If there is an error updating the transaction object, such as:
+   * If the transaction object to be updated does not exist in the state.
+   */
+  async updateTransaction(data: Transaction): Promise<void> {
+    try {
+      await this.update(async (state: SnapState) => {
+        const dataInState = await this.findTransaction(
+          {
+            txnHash: data.txnHash,
+            chainId: data.chainId,
+          },
+          state,
+        );
+        if (!dataInState) {
+          throw new Error(`Transaction does not exist`);
+        }
+        this.updateEntity(dataInState, data);
+      });
+    } catch (error) {
+      throw new StateManagerError(error.message);
+    }
+  }
+
+  /**
+   * Adds a new transaction object to the state with the given data.
+   *
+   * @param data - The transaction object to add.
+   * @returns A Promise that resolves when the add is complete.
+   * @throws {StateManagerError} If there is an error adding the transaction object, such as:
+   * If the transaction object to be added already exists in the state.
+   */
+  async addTransaction(data: Transaction): Promise<void> {
+    try {
+      await this.update(async (state: SnapState) => {
+        const dataInState = await this.findTransaction(
+          {
+            txnHash: data.txnHash,
+            chainId: data.chainId,
+          },
+          state,
+        );
+
+        if (dataInState) {
+          throw new Error(`Transaction already exist`);
+        }
+        state.transactions.push(data);
+      });
+    } catch (error) {
+      throw new StateManagerError(error.message);
+    }
+  }
+
+  /**
+   * Removes the transaction objects in the state with the given search conditions.
+   *
+   * @param searchFilter - The searchFilter object.
+   * @returns A Promise that resolves when the remove is complete.
+   */
+  async removeTransactions(searchFilter: SearchFilter): Promise<void> {
```

Comment:
What is use case to removeTransactions ?

Reply:
remove outdate txn in state, please refer to getTransations, at the end we remove txn from state, as we only display a limit of time of the txns